src/machines/java/com/enderio/machines/common/blockentity/CrafterBlockEntity.java

```diff
@@ -0,0 +1,170 @@
+    private static final int ENERGY_USAGE_PER_ITEM = 10;
+
+    private CraftingRecipe recipe;
+    private NonNullList<ItemStack> outputBuffer = NonNullList.create();
+
+    private static final CraftingContainer dummyCContainer = new CraftingContainer(new AbstractContainerMenu(null, -1) {
+        @Override
+        public ItemStack quickMoveStack(Player pPlayer, int pIndex) {
+            return ItemStack.EMPTY;
+        }
+
+        @Override
+        public boolean stillValid(Player pPlayer) {
+            return false;
+        }
+    }, 3, 3);
+
+
+    public CrafterBlockEntity(BlockEntityType<?> type, BlockPos worldPosition, BlockState blockState) {
+        super(EnergyIOMode.Input, ENERGY_CAPACITY, ENERGY_TRANSFER, ENERGY_USAGE, type, worldPosition, blockState);
+    }
+
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory inventory, Player player) {
+        return new CrafterMenu(this, inventory, containerId);
+    }
+
+    @Override
+    public MachineInventoryLayout getInventoryLayout() {
+        return MachineInventoryLayout
+            .builder()
+            .capacitor()
+            .setStackLimit(1)
+            .inputSlot(9, this::acceptSlotInput)
+            .setStackLimit(64)
+            .outputSlot(1)
+            .ghostSlot(9)
+            .previewSlot()
+            .build();
+    }
+
+    private boolean acceptSlotInput(int slot, ItemStack stack) {
+        return this.getInventory().getStackInSlot(slot + 10).sameItem(stack);
+    }
+
+    @Override
+    public void serverTick() {
+        processOutputBuffer();
+
+        Optional<ItemStack> opt = getRecipeResult();
+        if (opt.isPresent()) {
+            ItemStack result = opt.get();
+            this.getInventory().setStackInSlot(20, result);
+            if (shouldActTick() && hasPowerToCraft() && canMergeOutput(result) && outputBuffer.isEmpty()) {
+                craftItem();
+            }
+        } else {
+            this.getInventory().setStackInSlot(20, ItemStack.EMPTY);
+        }
+        super.serverTick();
+    }
+
+    private boolean shouldActTick() {
+        return canAct() && level.getGameTime() % ticksForAction() == 0;
+    }
+
+    private int ticksForAction() {
+        return 20;
+    }
+
+    private boolean hasPowerToCraft() {
+        return this.energyStorage.consumeEnergy(ENERGY_USAGE_PER_ITEM, true) > 0;
+    }
+
+    private void processOutputBuffer() {
+        if (outputBuffer.isEmpty()) {
+            return;
+        }
+        // clean buffer
+        outputBuffer.removeIf(ItemStack::isEmpty);
+
+        // output
+        if (canMergeOutput(outputBuffer.get(0))) {
+            var stack = getInventory().getStackInSlot(10);
+            if (stack.isEmpty()) {
+                getInventory().setStackInSlot(10, outputBuffer.get(0).copy());
+            } else if (stack.sameItem(outputBuffer.get(0))) {
+                stack.grow(1);
+            }
```

Comment:
just use else, because the canMergeOutput does this check implicitly
also grow by the size of the outputbuffer itemstack